### PR TITLE
bootstrap csctl-plugin-openstack repo

### PR DESCRIPTION
**Pull request by kranurag7**
this contains boilerplate for working with this repository.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+	@if [ -f csctl-plugin-openstack ]; then rm csctl-plugin-openstack; fi
+
+
+##@ Common
+##########
+# Common #
+##########
+.PHONY: build
+build: # build the csctl-plugin-openstack binary
+	go build -ldflags "$(LDFLAGS)" -o csctl-plugin-openstack main.go
```
Comment (michal-gubricky) [inline Makefile:60]: Do we need `"$(LDFLAGS)"` when it is not defined in this Makefile?

**Review by michal-gubricky — CHANGES_REQUESTED** [review]

File: .github/renovate/groups.json5 [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      matchManagers: ["gomod"],
+      ignorePaths: ["hack/tools/**"],
+      commitMessageTopic: "Golang Dependencies group",
+      groupSlug: "golang-deps",
+    },
+    {
+      description: "disable update of builder image",
+      matchManagers: ["github-actions"],
+      matchUpdateTypes: ["major", "minor", "patch"],
+      matchPackageNames: ["ghcr.io/sovereigncloudstack/builder"],
```
[the remark this answers is not in the document]
Comment (kranurag7) [reply .github/renovate/groups.json5:62]: I think overtime, if we develop in a lot of repos then we will end up having different builder images with similar content. 
We create a new repo to have a central builder image and using the same in at least the csctl and it's plugin repos. 
The builder image is built locally and pushed using 0.1.0 tag. 
Later we will automate this using renovate. 
Here's the PR for the same. https://github.com/SovereignCloudStack/builder/pull/1

File: Makefile (hunk 1 above)
Comment (kranurag7) [reply Makefile:61]: updated via https://github.com/SovereignCloudStack/csctl-plugin-openstack/commit/c441a6cd91808346c4ac5d1b6e1eecfdd1db5529

File: .github/renovate/groups.json5 (hunk 2 above)
[the remark this answers is not in the document]
Comment (michal-gubricky) [reply .github/renovate/groups.json5:62]: I am okay with the approach you described.

**Review by michal-gubricky — APPROVED** [review]

**Comment by michal-gubricky** [comment]
Can this PR be merged or do we want to add something here?

**Comment by kranurag7** [comment]
> Can this PR be merged or do we want to add something here?

@michal-gubricky From my side, this looks good, if something breaks, I can fix it in subsequent PRs for sure. 

I'll squash the commits and then we can proceed with merge.

**Pull request merged**